cpp/include/raft/handle.hpp

```diff
@@ -62,11 +63,23 @@ class handle_t {
         CUDA_CHECK(cudaGetDevice(&cur_dev));
         return cur_dev;
       }()),
-      num_streams_(n_streams),
+      streams_(n_streams),
       device_allocator_(std::make_shared<mr::device::default_allocator>()),
       host_allocator_(std::make_shared<mr::host::default_allocator>()) {
     create_resources();
   }
+  handle_t(const handle_t& h) : dev_id_(h.get_device()) {}
+  handle_t(const handle_t&& h) : dev_id_(h.get_device()) {}
+
+  // light copy operator
+  // skip streams, comms, and libs handles
+  handle_t& operator=(const handle_t& h) {
+    prop_ = h.get_device_properties();
+    device_prop_initialized_ = true;
+    device_allocator_ = get_device_allocator();
+    host_allocator_ = get_host_allocator();
+    return *this;
+  }
```

Comment:
If you have defined a move constructor, there should be a move operator as well.

And I feel like this is non-idiomatic to use copy/move constructors/operators for partial copies/moves (See the "rulle of five" https://cpppatterns.com/patterns/rule-of-five.html).

And typically, copy constructor can be implemented using { *this = other; } if a copy operator is defined, but here the behaviors of copy constructor & copy operator are different. This can be very surprising to some people.

Reply:
Thanks for noticing the move constructor operator. For completeness, I added these really because the compiler was complaining about it for the ROV in `get_handle_from_internal_pool`, but it does not end up being used.
I'll add this in my next commit. 

Answering the copy discussion in the other thread and resolving this one.